draft-romann-mud-constrained.md

```diff
@@ -180,22 +180,110 @@ In the following, we will first outline these additional means for exposing MUD
+This allows for MUD receivers or MUD controllers to verify the authenticity of the provided MUD-URL.
+
+CBOR Web Tokens that contain MUD-URL information have the following properties:
+- The MUD-URL is contained as an ASCII-encoded string in the "mud-url" claim.
+- The Token MAY contain Proof-of-Possession claims {{!RFC8747}}.
+  If it does, the MUD receiver MUST verify that the device is in possession of the key specified in the cnf claim.
+  Proof-of-Possession claims that are transmitted over CoAP without DTLS/TLS MUST represent public keys, Receivers MUST treat any symmetric keys sent over insecure channels as invalid/compromised.
+  <!-- TODO specify PoP mechanism in flow section -->
+- The Token MAY contain an expiry time.
+  If an expiry time is specified, the MUD-URL should be resubmitted or requested again shortly before the original CWT expires.
+  Note that using an expiry time could cause problems if the device is unable to perform a refresh, e.g., due to a power outage.
+  <!-- TODO maybe be more specific regarding the time where the refresh should happen -->
+
+CoAP requests and responses that use this format MUST use the Content-Format option with the value corresponding to the "application/mud-url+cwt" media type.
+
+## Proof-of-Possession for CBOR Web Tokens with MUD-URLs {#general_pop}
+If the MUD-URL is transmitted as a CBOR Web Token that includes Proof-of-Possession claims, CoAPS MUST be used to perform proof-of-possession.
+To do so, the following procedures SHOULD be used.
+Both of the following procedures use the establishment of a DTLS session using the PoP key in order to prove the possession of the key, similarly to the procedures defined in {{!RFC9202}}.
+
+<!-- TODO required ciphers? -->
+### For the Thing-initiated Flow
+1.  After submitting the MUD-URL, the MUD Receiver parses the token.
+    If it detects Proof-of-Possession claims, the receiver MUST reply with a 4.01 (Unauthorized) CoAP response code and reject the token for now.
+    If the original submission request was not performed using CoAPS, the Receiver MUST also return a set of key information that can be used to establish a CoAP+DTLS session with it, as well as a CoAPS URI indicating the location of the secured submission resource.
+    <!-- TODO key format? -->
+2.  The Thing then repeats the submission request using its own key information, the Receiver's key information, and the provided URI.
+3.  During the DTLS handshake, the Receiver MUST require the Thing to use the Proof-of-Possession key included in the originally submitted token in order to successfully establish a DTLS session.
+4.  After the DTLS handshake, the Thing has proven possession of the key included in the CWT to the MUD Receiver, which can then treat the CWT as valid.
+<!-- TODO do we still want to re-send the CWT? -->
```

Comment:
Hmm, in the rendered view, this subsection is a bit broken formatting-wise.

![image](https://user-images.githubusercontent.com/12641361/221200241-74737cdb-797b-4a2b-af09-bb1f01ce7b64.png)